src/xdp/program.c

```diff
@@ -44,7 +44,11 @@ typedef struct _XDP_PROGRAM_FRAME_STORAGE {
         IPV4_HEADER Ip4Hdr;
         IPV6_HEADER Ip6Hdr;
     };
-    UDP_HDR UdpHdr;
+    union {
+        UDP_HDR UdpHdr;
+        TCP_HDR TcpHdr;
+    };
+    UINT8 TcpHdrOptions[40]; // Up to 40B options/paddings
```

Comment:
Are we using TcpHdrOptions at all? If not, let's just ignore it. We're definitely not in the business of validating header correctness, just whatever minimal checks are required to ensure our reads are in bounds.

Reply:
We are not using it. The reason I have it there is because calling XdpGetContiguousHeader was the most convenient way for me to do the checks and advance fragment index, offset and etc. XdpGetContiguousHeader requires passing a storage.

Reply:
Makes sense. If we're not using the field, then we could remove all of this, right? Do we think we're likely to add new rules that would need to inspect TCP options?

```
    if (TCP_HDR_LEN_TO_BYTES(Cache->TcpHdr->th_len) > sizeof(Storage->TcpHdr)) {
        //
        // Attempt to read TCP options.
        //
        Valid =
            XdpGetContiguousHeader(
                Frame, Buffer, BufferDataOffset, FragmentIndex, FragmentsRemaining, FragmentRing,
                VirtualAddressExtension,
                &Storage->TcpHdrOptions,
                TCP_HDR_LEN_TO_BYTES(Cache->TcpHdr->th_len) - sizeof(Storage->TcpHdr),
                &Cache->TcpHdrOptions);
    }
```

Reply:
I don't think we will ever touch TCP options but we still need to advance fragments to the boundary of payload because we need to parse the TCP payload for CIBIR rules in the near future, right?

Reply:
Ah, yes, I forgot we do care about the TCP payload. Dang...

We could create a function that advances over fragments without copying data, but that introduces even more code we need to maintain without substantially simplifying parsing. Since we expect the common case to be unfragmented anyways, this PR's approach is probably the path of least resistance.

Reply:
Kept options handling the logic as is.